code/game/machinery/_machinery.dm

```diff
@@ -188,15 +191,22 @@
  * proc to call when the machine starts to require power after a duration of not requiring power
  * sets up power related connections to its area if it exists and becomes area sensitive
  * does not affect power usage itself
+ *
+ * Returns TRUE if it triggered a full registration, FALSE otherwise
+ * We do this so machinery that want to sidestep the area sensitiveity optimization can
  */
 /obj/machinery/proc/setup_area_power_relationship()
-	become_area_sensitive(INNATE_TRAIT)
-
 	var/area/our_area = get_area(src)
 	if(our_area)
 		RegisterSignal(our_area, COMSIG_AREA_POWER_CHANGE, .proc/power_change)
+
+	if(HAS_TRAIT(src, TRAIT_AREA_SENSITIVE)) // If we for some reason have not lost our area sensitivity, there's no reason to set it back up
```

Comment:
Shouldn't this check for area sensitive from innate source only ?

Reply:
good point yes